Sources/Nimble/Matchers/BeLogical.swift

```diff
@@ -141,10 +141,13 @@ extension NMBObjCMatcher {
         }
     }
 
-    @objc public class func beFalseMatcher() -> NMBObjCMatcher {
-        return NMBObjCMatcher(canMatchNil: false) { actualExpression, failureMessage in
-            let expr = actualExpression.cast { ($0 as? NSNumber)?.boolValue ?? false }
-            return try beFalse().matches(expr, failureMessage: failureMessage)
+    @objc public class func beFalseMatcher() -> NMBMatcher {
+        return NMBPredicate { actualExpression in
+            let expr = actualExpression.cast { value -> Bool? in
+                guard let value = value else { return nil }
```

Comment:
`nil` should not be converted to `false` here.

Reply:
Oh, good catch. Should we add a test that covers this difference between `beFalse()` and `beFalsy()`? We're also accepting values that fail to cast to `NSNumber`, should we also return nil if the cast fails?

Reply:
That is already covered here: https://github.com/Quick/Nimble/blob/7de081475abd9db7e43355990c8f8ec0c178efe3/Tests/NimbleTests/objc/ObjCBeFalseTest.m#L24-L29

Previously the behavior is ensured by `NMBObjCMatcher(canMatchNil: false)`: https://github.com/Quick/Nimble/blob/7de081475abd9db7e43355990c8f8ec0c178efe3/Sources/Nimble/Adapters/NMBObjCMatcher.swift#L47-L52

Reply:
What I wanted to say that is that the message

> `nil` should not be converted to `false` here.

is only applicable to `NMBPredicate` in this PR, but not to the previous `NMBObjCMatcher` impl.